syntax = "proto3";

import public "common.proto";

package dyntrace.proto.process;

message list_tracepoint
{
}
message add_tracepoint
{
    oneof location
    {
        string filter = 1;
        string regex = 2;
        uint64 address = 3;
    };
    bool entry_exit = 4;
    string tracer = 5;
    repeated string tracer_args = 6;
    string name = 7;
}
message remove_tracepoint
{
    string name = 1;
    repeated uint32 id = 2;
}
message list_symbol
{
}

message hello
{
}

message request
{
    oneof req
    {
        list_tracepoint list_tp = 1;
        add_tracepoint add_tp = 2;
        remove_tracepoint remove_tp = 3;
        hello hello = 4;
        list_symbol list_sym = 5;
    }
}

message process_message
{
    uint64 seq = 1;
    oneof body
    {
        request req = 2;
        response resp = 3;
    }
}